stdlib/public/core/CTypes.swift

```diff
@@ -101,6 +101,11 @@ public typealias CLongDouble = Double
 #endif
 #elseif os(OpenBSD)
 public typealias CLongDouble = Float80
+#elseif os(FreeBSD)
+#if arch(x86_64) || arch(i386)
+public typealias CLongDouble = Float80
+#endif
+// TODO: Fill in definitions for additional architectures as needed.
```

Comment:
This should alias to `Double` here.  I think we might as well as do that now rather than wait for that.

Reply:
For arm64-freebsd, at least, `long double` is float128, which would be different from `Double`.  This is similar to what's going on in the Linux branch above:

```
// TODO: Fill in definitions for additional architectures as needed. IIRC
// armv7 should map to Double, but arm64 and ppc64le should map to Float128,
// which we don't yet have in Swift.
```

I figured I'd match that in spirit, no?

Reply:
(As a general review point, It also means there is a sensible default or #error rather than causing an error implicitly for a missing type alias. The suggestion to use Double suffices, though.)

Reply:
In this case it would be even worse, it would be a weird error when trying to import a C function that uses a `long double` in its signature.  So, what @3405691582 points out is pretty critical.  I think that just handling the case properly with the alias to `Double` (which is FP64 and thus matches the ABI) is the proper thing to do.

Reply:
Agreed—the _weird error_ is precisely what I hit!  I've updated this PR to `#error` in cases where `CLongDouble` isn't defined.

> I think that just handling the case properly with the alias to Double (which is FP64 and thus matches the ABI) is the proper thing to do.

`Double` doesn't always match, though.  For example, `freebsd-arm64` uses float128 for `long double` (see my `__LDBL_MANT_DIG__` experiment in a different comment).